File: api/toggle.py
from http.server import BaseHTTPRequestHandler
import json
import requests

# Your Credentials
API_KEY = "65cfb3f31b90b2e9b429422f8781660a1a2a08b9"
CONFIG_ID = "d6ddb9"
DOMAIN = "ppq.apple.com"

class handler(BaseHTTPRequestHandler):
    def do_GET(self):
        url = f"https://api.nextdns.io/profiles/{CONFIG_ID}/denylist"
        headers = {"X-Api-Key": API_KEY, "Content-Type": "application/json"}

        # 1. Check if domain is already blocked
        response = requests.get(url, headers=headers)
        data = response.json().get('data', [])
        is_blocked = any(item.get('id') == DOMAIN for item in data)

        if is_blocked:
            # 2. If blocked, DELETE (Unblock)
            delete_url = f"{url}/{DOMAIN}"
            requests.delete(delete_url, headers=headers)
            status = "Unblocked"
        else:
            # 3. If not blocked, POST (Block)
            requests.post(url, headers=headers, json={"id": DOMAIN})
            status = "Blocked"

        self.send_response(200)
        self.send_header('Content-type', 'application/json')
        self.end_headers()
        self.wfile.write(json.dumps({"status": "success", "action": status}).encode())

File: api/toggle.js
export const config = { runtime: 'edge' };

export default async function handler(req) {
  const NEXTDNS_PROFILE_ID = process.env.NEXTDNS_PROFILE_ID;
  const NEXTDNS_API_KEY = process.env.NEXTDNS_API_KEY;
  const DOMAIN = "ppq.apple.com";

  const url = `https://api.nextdns.io/profiles/${NEXTDNS_PROFILE_ID}/denylist`;
  const headers = { "X-Api-Key": NEXTDNS_API_KEY, "Content-Type": "application/json" };

  try {
    // 1. Check current status
    const response = await fetch(url, { headers });
    const data = await response.json();
    
    const exists = data.data.find(item => item.id === DOMAIN);

    if (exists) {
      // 2. If it exists, DELETE it (Turn OFF blocking)
      await fetch(`${url}/${DOMAIN}`, { method: 'DELETE', headers });
      return new Response(`UNBLOCKED: ${DOMAIN}`);
    } else {
      // 3. If it doesn't exist, POST it (Turn ON blocking)
      await fetch(url, {
        method: 'POST',
        headers,
        body: JSON.stringify({ id: DOMAIN, active: true })
      });
      return new Response(`BLOCKED: ${DOMAIN}`);
    }
  } catch (err) {
    return new Response(`Error: ${err.message}`, { status: 500 });
  }
}
